typescript-lite/generator/src/main/java/org/coursera/courier/tslite/TSProperties.java

```diff
@@ -0,0 +1,67 @@
+package org.coursera.courier.tslite;
+
+/**
+ * Customizable properties that may be added to a Pegasus schema.
+ *
+ * Example usage:
+ *
+ * <code>
+ *   {
+ *     "name": "Fortune",
+ *     "namespace": "org.example",
+ *     "type": "record",
+ *     "fields": [ ... ],
+ *     "typescript": {
+ *       "optionality": "STRICT"
+ *     }
+ *   }
+ * </code>
+ */
+public class TSProperties {
+
+  /**
+   * "optionality" property.
+   *
+   * Typescript representations of Pegasus primitive types supported by this generator.
+   */
+  public enum Optionality {
+
+    /**
+     * Allows required fields to be absent, useful when working with projections.
+     *
+     * "undefined" is used to represent un-projected fields (required or optional) as well as absent
+     * optional fields.
+     */
+    REQUIRED_FIELDS_MAY_BE_ABSENT,
+
+    // TODO(jbetz): Remove as soon as we've migrated away from this usage pattern.
```

Comment:
Actually I'm curious about it too. Is Courier getting rid of language-level optionality? The TODO was part of the sister file in the `swift-generator` file. Maybe Joe knows?